controllers/workspace/finalize.go

```diff
@@ -223,10 +225,8 @@ func (r *DevWorkspaceReconciler) getClusterCleanupJob(ctx context.Context, works
 	return clusterJob, nil
 }
 
-func isFinalizerNecessary(_ *v1alpha2.DevWorkspace) bool {
-	// TODO: Implement checking whether persistent storage is used (once other choices are possible)
-	// Note this could interfere with cloud-shell until this TODO is resolved.
-	return true
+func isFinalizerNecessary(workspace *v1alpha2.DevWorkspace) bool {
+	return storagelib.NeedsStorage(workspace.Spec.Template)
```

Comment:
They should still be cleaned up because we never remove the finalizer -- presumably it'll be set the first time the workspace needs storage, and will only be removed once the PVC job has run.